tests/unit/service/test_engine.py

```diff
@@ -219,3 +220,59 @@ def test_engine_new_status(
     engine = resource_manager.engines.get_by_name(mock_engine.name)
 
     assert engine.current_status == expected_status
+
+
+@mark.parametrize(
+    "spec_input, expected_spec, status_input, expected_status",
+    [
+        # Test all valid InstanceType values
+        ("S", InstanceType.S, "RUNNING", EngineStatus.RUNNING),
+        ("M", InstanceType.M, "STOPPED", EngineStatus.STOPPED),
+        ("L", InstanceType.L, "STARTING", EngineStatus.STARTING),
+        ("XL", InstanceType.XL, "STOPPING", EngineStatus.STOPPING),
+        # Test InstanceType enum values directly
+        (InstanceType.S, InstanceType.S, "FAILED", EngineStatus.FAILED),
+        (InstanceType.M, InstanceType.M, "REPAIRING", EngineStatus.REPAIRING),
+        # Test unknown/invalid values that should default to UNKNOWN
+        ("INVALID_TYPE", InstanceType.UNKNOWN, "INVALID_STATUS", EngineStatus.UNKNOWN),
+        ("XXL", InstanceType.UNKNOWN, "WEIRD_STATE", EngineStatus.UNKNOWN),
```

Comment:
These 2 do not exist today, but they can. We also have type families, not sure whether SDKs care, though.